src/predictor/cpu_predictor.cc

```diff
@@ -344,7 +391,9 @@ class CPUPredictor : public Predictor {
         }
         feats.Fill(page[i]);
         for (unsigned j = 0; j < ntree_limit; ++j) {
-          int tid = model.trees[j]->GetLeafIndex(feats);
+          auto const& tree = *model.trees[j];
+          auto const& cats = tree.GetCategoriesMatrix();
+          bst_node_t tid = GetLeafIndex<true, true>(tree, feats, cats);
```

Comment:
Should we also add a check for `has_categorical` here? Or are we purposefully removing it as `GetLeaf` is not performance critical?

Reply:
also it's possible `feats` has no missing values

Reply:
Yes, I'm assuming it's not critical.  Otherwise we will have a lot more specializations

Reply:
This is the previous default so I'm not slowing it down here.  If optimization is needed (being the computation bottleneck of some algorithms/models), we can come back to it in a different PR that focuses on optimization.